bl2/bl2_image_load_v2.c

```diff
@@ -88,7 +88,7 @@ entry_point_info_t *bl2_load_images(void)
 	assert(bl2_to_next_bl_params->head->ep_info);
 
 	/* Populate arg0 for the next BL image if not already provided */
-	if (bl2_to_next_bl_params->head->ep_info->args.arg0 == 0)
+	if (bl2_to_next_bl_params->head->ep_info->args.arg0 == (u_register_t)0)
```

Comment:
Should fix the issue.

Reply:
You could also write like follows if you want to avoid the MISRA issue.

    if (!bl2_to_next_bl_params->head->ep_info->args.arg0)

Reply:
Unfortunately, that would trigger the original issue: 

MISRA C-2012 Rule 10.1 violation: The operand "bl2_to_next_bl_params->head->ep_info->args.arg0" has essential type "unsigned 64-bit int", but should be Essentially Boolean.

Reply:
Please skip except mandatory.
This check is ridiculous.

Reply:
For what it's worth, I think a better (and more consistent) fix for this would have been to use `U(0)` or `0u` in the comparison instead of the cast.

Both 10.1 and 10.4 are "required" MISRA rules so we'd like to keep them on if possible. However, we haven't properly announced our policy yet. We should do this.